index.js

```diff
@@ -54,6 +59,17 @@ module.exports = function * (next) {
     }
   }
 
+  //Adjust infinite end
+  if (end == Number.POSITIVE_INFINITY) {
+    if(isNaN(len)) {
```

Comment:
This is not the same thing. We only want to reassign the end if end is Inf and we know the length.